extensions/positron-catalog-explorer/src/catalogs/databricks.ts

```diff
@@ -0,0 +1,605 @@
+		}
+
+		session.execute(
+			code,
+			session.runtimeMetadata.languageId,
+			positron.RuntimeCodeExecutionMode.Interactive,
+			positron.RuntimeErrorBehavior.Continue,
+		);
+	}
+
+	private async uriWithWarehouse(
+		node: CatalogNode,
+	): Promise<vscode.Uri | undefined> {
+		if (!node.resourceUri) {
+			return;
+		}
+		if (node.type === 'table') {
+			await this.chooseWarehouse();
+			if (!this.warehousePath) {
+				return;
+			}
+			return withHttpPath(node.resourceUri, this.warehousePath);
+		}
+		return node.resourceUri;
+	}
+
+	private async chooseWarehouse() {
+		if (this.warehousePath) {
+			return;
+		}
+		const warehouses = await getSqlWarehouses(this.workspace, this.token);
+		const items = warehouses.map((v) => {
+			return {
+				item: v,
+				label: v.name,
+				description: `ID: ${v.id}`,
+				detail: `[${v.warehouse_type}] SQL Warehouse`,
+			};
+		});
+		const choice = await vscode.window.showQuickPick(items, {
+			title: 'Choose an SQL Warehouse for the Connection',
+		});
+		if (!choice) {
+			return undefined;
+		}
+		this.warehousePath = choice.item.odbc_params.path;
+	}
+}
+
+function getCodeForUri(
+	languageId: string,
+	uri: vscode.Uri,
+	type: CatalogNodeType,
+): { code: string; dependencies: string[] } {
+	switch (languageId + '_' + type) {
+		case 'python_file':
+			return getPythonCodeForFile(uri);
+		case 'python_table':
+			return getPythonCodeForTable(uri);
+		case 'r_file':
+			return getRCodeForFile(uri);
+		case 'r_table':
+			return getRCodeForTable(uri);
+		default:
+			throw new Error(
+				`Code generation for language '${languageId}' and type '${type}' is not yet supported`,
+			);
+	}
+}
+
+function getPythonCodeForFile(uri: vscode.Uri): {
+	code: string;
+	dependencies: string[];
+} {
+	const dependencies = ['databricks-sdk', 'pandas'];
+	const ext = path.extname(uri.path);
+	const varname = nameToIdentifier(path.basename(uri.path).replace(ext, ''));
+	const code = `# pip install ${dependencies.join(' ')}
+
+import pandas as pd
+from databricks.sdk import WorkspaceClient
+from io import BytesIO
+
+w = WorkspaceClient(host="https://${uri.authority}")
+${varname} = pd.read_csv(
+	BytesIO(w.files.download("${uri.path}").contents.read())
+)
+`;
+	return { code, dependencies };
+}
+
+function getPythonCodeForTable(uri: vscode.Uri): {
+	code: string;
+	dependencies: string[];
+} {
+	const params = new URLSearchParams(uri.query);
+	const catalog = params.get('catalog');
+	const schema = params.get('schema');
+	const table = uri.path.replace(/^\//, '');
+	const httpPath = params.get('http_path');
+	if (!catalog || !schema || !httpPath) {
+		throw new Error('Malformed Databricks table URI');
+	}
+	const dependencies = ['databricks-sql-connector', 'pyarrow', 'pandas'];
+	const varname = nameToIdentifier(table);
+	const code = `# pip install ${dependencies.join(' ')}
+
+import pandas as pd
+from databricks import sql
+
+conn = sql.connect(
+	server_hostname="https://${uri.authority}",
```

Comment:
super nit: I think these got changed to leading tabs instead of spaces, and I prefer spaces for Python code

Reply:
Wanted to make sure you didn't miss this but it's totally fine if you want to keep the tabs!